src/iexamine.cpp

```diff
@@ -2280,6 +2280,167 @@ void iexamine::tree_hickory(player &p, const tripoint &examp)
+            }
+        }
+    }
+
+    enum options {
+        REMOVE_TAP,
+        ADD_CONTAINER,
+        HARVEST_SAP,
+        REMOVE_CONTAINER,
+        CANCEL,
+    };
+    uimenu selectmenu;
+    selectmenu.addentry( REMOVE_TAP, true, MENU_AUTOASSIGN, _("Remove tap") );
+    selectmenu.addentry( ADD_CONTAINER, !has_container, MENU_AUTOASSIGN, _("Add a container to receive the %s"), maple_sap_name.c_str() );
+    selectmenu.addentry( HARVEST_SAP, has_sap, MENU_AUTOASSIGN, _("Harvest current %s (%d)"), maple_sap_name.c_str(), charges );
+    selectmenu.addentry( REMOVE_CONTAINER, has_container, MENU_AUTOASSIGN, _("Remove container") );
+    selectmenu.addentry( CANCEL, true, MENU_AUTOASSIGN, _("Cancel") );
+
+    selectmenu.return_invalid = true;
+    selectmenu.text = _("Select an action");
+    selectmenu.selected = 0;
+    selectmenu.query();
+
+    switch( static_cast<options>( selectmenu.ret ) ) {
+        case REMOVE_TAP: {
+            if( !p.has_items_with_quality( "HAMMER", 1, 1 ) ) {
+                add_msg( m_info, _( "You need a hammering tool to remove the spile from the crust." ) );
+                return;
+            }
+
+            const std::string spile_name = item( "maple_tree_spile", 0 ).tname( 1 );
+            item maple_tree_spile( "maple_tree_spile", calendar::turn );
+            add_msg( _( "You remove the %s." ), spile_name.c_str() );
+            g->m.add_item_or_charges( p.pos(), maple_tree_spile );
+
+            if ( has_container ) {
+                for( auto &it : items ) {
+                    g->m.add_item_or_charges( p.pos(), it );
+                }
+            }
+            g->m.i_clear( examp );
+
+            p.mod_moves( -200 );
+            g->m.ter_set( examp, t_tree_maple );
+
+            return;
+        }
+
+        case ADD_CONTAINER: {
+            auto cont_loc = maple_tree_sap_container();
+
+            item *container = cont_loc.get_item();
+            if( container ) {
+                g->m.add_item_or_charges( examp, *container, 0 );
+
+                cont_loc.remove_item();
+            } else {
+                add_msg( m_info, _( "No container added. The sap will just spill on the ground." ) );
+            }
+
+            return;
+        }
+
+        case HARVEST_SAP:
+            for( auto &it : items ) {
+                if( it.is_bucket() || it.is_watertight_container() ) {
+                    auto &liquid = it.contents.front();
+                    if( !it.is_container_empty() && liquid.type->id == "maple_sap" ) {
+                        long initial_charges = liquid.charges;
+                        bool emptied = g->handle_liquid( liquid, false, false, &it, NULL, PICKUP_RANGE );
+
+                        if( emptied || initial_charges != liquid.charges ) {
+                            p.mod_moves( -100 );
+                        }
+
+                        if( emptied || liquid.charges <= 0 ) {
+                            it.contents.pop_back();
```

Comment:
You're popping the back, but checking the front. While at the moment liquid containers can't contain more than one item (so `back() == front()`), it would be better to `erase( contents.begin() )`, `contents.clear()` or something like that.